src/config/settings.js

```diff
@@ -8,27 +8,38 @@ const fs = require('fs');
 const util = require('util');
 const { dirname } = require('path');
 const deepmerge = require('deepmerge');
+const deepequal = require('deep-equal')
 
-// Load the settings file
-function load() {
-
-  // read the settings file into memory
-  const rawdata = fs.readFileSync(paths.config() + '/settings.json');
-  let settings = JSON.parse(rawdata);
-  let cfg = deepmerge(defaultSettings, settings);
-
-  // TODO: validate it
+// Settings versions are independent of app versions.
+// Any time a version changes, increment its version by 1.
+// Adding a new setting usually doesn't require incrementing the version field, because
+// it will be picked up from the default settings object.
+// Version incrementing is for when a breaking change is introduced in the settings object.
 
-  return cfg;
-
-}
+const CURRENT_SETTINGS_VERSION = 1;
 
 const defaultSettings = {
+  version: CURRENT_SETTINGS_VERSION,
   kubernetes: {
     version: "v1.19.2"
   }
 }
 
+// Load the settings file
+function load(inBrowser=false) {
```

Comment:
We should probably just drop `inBrowser`, since it'll never be true.  But I'm okay with it not being done in this PR.

Further refactoring idea: make this `async`, so we can stop doing `readFileSync`. Also not in this PR :)

Reply:
Dropped inBrowser deletion here rather than create a new issue. 

As for the `readFileSync`, I don't see a problem doing a sync read in a UI program on a file that I expect will never surpass 10K bytes. My philosophy because node was originally designed for servers, where you want to do all file access async so that you can serve multiple clients in parallel. Here there's only the one client, sitting at a keyboard most of the time.